app/controllers/items_controller.rb

```diff
@@ -49,16 +50,36 @@ def create
   end
 
   def destroy
-    if @item.destroy
-      flash[:success] = "Item '#{@item.description}' deleted!"
+    if @item.deleted?
+      flash[:error] = "'#{@item.description}' is already deleted."
     else
-      flash[:error] = "'#{@item.description}' could not be deleted."
+      @item.deleted_at = Time.zone.now
+      if @item.save!
+        flash[:success] = "Item '#{@item.description}' deleted!"
+      else
+        flash[:error] = "'#{@item.description}' was unable to be deleted."
+      end
     end
     redirect_to items_path(category_id: @item.category_id)
   end
 
+  def restore
+    @item.restore
+
+    redirect_to edit_item_path(@item)
+  end
+
+  def deleted
+    @items = Item.all.deleted
+    @category = "Deleted Items"
+  end
+
   private
 
+  def notify_if_deleted
+    flash[:warning] = "'#{@item.description}' is deleted." if @item.deleted?
```

Comment:
The `flash` is typically for completed actions or to communicate something that happened rather than existing state of an object. I think this belongs in the view.